core/storage/suggestion/gae_models.py

```diff
@@ -2289,34 +2289,49 @@ def fetch_page(
 
         sorted_results: List[
             TranslationSubmitterTotalContributionStatsModel] = []
+        filtered_results: List[
+            TranslationSubmitterTotalContributionStatsModel] = []
         today = datetime.date.today()
+        more: bool
+        next_result_model: Sequence[
+            TranslationSubmitterTotalContributionStatsModel] = []
 
         if num_days_since_last_activity is not None:
             last_date = today - datetime.timedelta(
                 days=num_days_since_last_activity)
-            next_offset = offset
-            while len(sorted_results) < page_size:
+            temp_offset = 0
+            while len(filtered_results) < page_size:
                 result_models: Sequence[
                     TranslationSubmitterTotalContributionStatsModel] = (
                     sort_query.fetch(
-                        NUM_MODELS_PER_FETCH, offset=next_offset))
+                        NUM_MODELS_PER_FETCH, offset=temp_offset))
                 if not result_models:
                     break
                 for result_model in result_models:
-                    next_offset += 1
+                    temp_offset += 1
                     if result_model.last_contribution_date >= last_date:
-                        sorted_results.append(result_model)
-                        if len(sorted_results) == page_size:
+                        filtered_results.append(result_model)
+                        if len(filtered_results) == page_size + offset:
                             break
+            sorted_results = filtered_results[offset:]
```

Comment:
The reason we have to compute them is because the last_activity filter is not directly applied in the gae query. We need to fetch the results again and discard the one that has already been sent to frontend.

Reply:
But that won't work due to the inefficiency I mentioned. Why can't you memorize the previous offset and then start counting from that when making the next query?

Reply:
Yeah, that was exactly what I did the first time I implemented it but while testing I mistakenly took the number of models to skip as same as offset and then created a lot of confusion which resulted in unneccessary efforts.